alc/backends/pulseaudio.cpp

```diff
@@ -783,6 +783,30 @@ void PulsePlayback::sinkInfoCallback(pa_context*, const pa_sink_info *info, int
         return;
     }
 
+    if(!mDevice->Flags.get<SampleTypeRequest>())
+    {
+        switch(info->sample_spec.format)
+        {
+        case PA_SAMPLE_U8:
+            mDevice->FmtType = DevFmtUByte;
+            break;
+        case PA_SAMPLE_S16NE:
+            mDevice->FmtType = DevFmtShort;
+            break;
+        case PA_SAMPLE_S32NE:
+            mDevice->FmtType = DevFmtInt;
+            break;
+        case PA_SAMPLE_FLOAT32NE:
+            mDevice->FmtType = DevFmtFloat;
+            break;
+        default:
+            break;
+        }
+    }
+
+    if(!mDevice->Flags.get<FrequencyRequest>())
+        mDevice->Frequency = info->sample_spec.rate;
```

Comment:
The `PA_STREAM_FIX_RATE` flag should already handle this (and properly resize the buffer, as well). I'm also not sure about setting the sample type because the sink device will typically be `s16le`, but the server-side mixing, volume control, and any other module processing could benefit from having floating-point samples prior to 16-bit quantization.

Reply:
Ok, I'll remove that...